pragma-node/src/infra/repositories/entry_repository.rs

```diff
@@ -44,6 +46,25 @@ pub struct MedianEntryRaw {
     pub num_sources: i64,
 }
 
+// Extended struct with components (non-optional)
+#[derive(Serialize, QueryableByName, Clone, Debug)]
+pub struct MedianEntryRawWithComponents {
+    #[diesel(sql_type = diesel::sql_types::Timestamptz)]
+    pub time: NaiveDateTime,
+
+    #[diesel(sql_type = diesel::sql_types::Numeric)]
+    pub median_price: BigDecimal,
+
+    #[diesel(sql_type = diesel::sql_types::BigInt)]
+    pub num_sources: i64,
+
+    #[diesel(sql_type = diesel::sql_types::Array<Record<(diesel::sql_types::Text, diesel::sql_types::Numeric, diesel::sql_types::Timestamptz)>>)]
+    pub components: Vec<(String, BigDecimal, NaiveDateTime)>,
+
+    #[diesel(sql_type = diesel::sql_types::Bool)]
```

Comment:
whats the point of this param, we will just know directly from the number of elements in the vec anyways ?